src/components/Books.js

```diff
@@ -102,8 +139,8 @@ useEffect(() => {
                    published ={item.volumeInfo.publishedDate}
                    image={item.volumeInfo.imageLinks.thumbnail ? item.volumeInfo.imageLinks.thumbnail : "N/A"}
                    link={item.volumeInfo.previewLink}
-                   bookAdded ={books && books[0].map(book => book.title )}
-                   bookId = {books && books[0].map(book => book.id)}
+                   bookAdded ={books[0] && books[0].map(book => book.book.title )}
```

Comment:
Why is this books[0] ? Do you mean to check the length of the books ? Is so you can use books.length. If not, you don't have to store the root level data in single array element format.

Reply:
I was saving the api response directly to the state. I will update it.